tests/integration/ux.test.ts

```diff
@@ -0,0 +1,240 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import fs, { createReadStream } from "fs";
+import minimist from "minimist";
+import readline from "readline";
+import { By, until, WebDriver } from "selenium-webdriver";
+import http from "http";
+
+import {
+  extensionLogsPresent, findAndAct,
+  getChromeDriver,
+  getFirefoxDriver, WAIT_FOR_PROPERTY
+} from "./utils";
+
+
+const args = minimist(process.argv.slice(2));
+console.debug(args);
+for (const arg of ["test_browser", "load_extension", "headless_mode"]) {
+  if (!(arg in args)) {
+    throw new Error(`Missing required option: --${arg}`);
+  }
+}
+
+const testBrowser = args["test_browser"];
+const loadExtension = args["load_extension"] === "true";
+const headlessMode = args["headless_mode"] === "true";
+
+export let webDriver;
```

Comment:
I think this should be called something like `getDriver` instead, to help differentiate from the `WebDriver` type import and the `driver` global.

Also, is this export being used somewhere?

Reply:
Yeah, this is used below to choose the right underlying driver (Firefox or Chrome) based on command-line flags. It would be nice to move this to utils or something and make it easier to share across repos.